packages/cli/src/commands/add.ts

```diff
@@ -27,156 +27,163 @@ const addOptionsSchema = z.object({
+          )
+          process.exit(1)
         }
 
-        if (!existsSync(targetDir)) {
-          await fs.mkdir(targetDir, { recursive: true })
-        }
+        continue
+      }
 
-        const existingComponent = item.files.filter((file) =>
-          existsSync(path.resolve(targetDir, file.name))
-        )
+      for (const file of item.files) {
+        let filePath = path.resolve(targetDir, file.name)
 
-        if (existingComponent.length && !options.overwrite) {
-          if (selectedComponents.includes(item.name)) {
-            logger.warn(
-              `Component ${item.name} already exists. Use ${chalk.green(
-                "--overwrite"
-              )} to overwrite.`
-            )
-            process.exit(1)
-          }
+        // Run transformers.
+        const content = await transform({
+          filename: file.name,
+          raw: file.content,
+          config,
+          baseColor,
+        })
 
-          continue
+        if (!config.tsx) {
+          filePath = filePath.replace(/\.tsx$/, ".jsx")
         }
 
-        for (const file of item.files) {
-          let filePath = path.resolve(targetDir, file.name)
-
-          // Run transformers.
-          const content = await transform({
-            filename: file.name,
-            raw: file.content,
-            config,
-            baseColor,
-          })
+        await fs.writeFile(filePath, content)
+      }
 
-          if (!config.tsx) {
-            filePath = filePath.replace(/\.tsx$/, ".jsx")
+      // Install dependencies.
+      if (item.dependencies?.length) {
+        const packageManager = await getPackageManager(cwd)
+        await execa(
+          packageManager,
+          [packageManager === "npm" ? "install" : "add", ...item.dependencies],
+          {
+            cwd,
           }
-
-          await fs.writeFile(filePath, content)
-        }
-
-        // Install dependencies.
-        if (item.dependencies?.length) {
-          const packageManager = await getPackageManager(cwd)
-          await execa(
-            packageManager,
-            [
-              packageManager === "npm" ? "install" : "add",
-              ...item.dependencies,
-            ],
-            {
-              cwd,
-            }
-          )
-        }
+        )
       }
-      spinner.succeed(`Done.`)
-    } catch (error) {
-      handleError(error)
     }
-  })
+    spinner.succeed(`Done.`)
+  } catch (error) {
+    handleError(error)
+  }
+}
+
+export const add = new Command()
+  .name("add")
+  .description("add a component to your project")
+  .argument("[components...]", "the components to add")
+  .option("-y, --yes", "skip confirmation prompt.", false)
+  .option("-o, --overwrite", "overwrite existing files.", false)
+  .option(
+    "-c, --cwd <cwd>",
+    "the working directory. defaults to the current directory.",
+    process.cwd()
+  )
+  .option("-p, --path <path>", "the path to add the component to.")
+  .action(runAdd)
```

Comment:
I didn't touch the core logic at all. Just extracted the core functionality to the `runAdd` and calling the func here